src/main/java/mil/dds/anet/search/pg/PostgresqlLocationSearcher.java

```diff
@@ -18,6 +18,13 @@ protected void addTextQuery(LocationSearchQuery query) {
     addFullTextSearch("locations", query.getText(), query.isSortByPresent());
   }
 
+  @Override
+  protected void addWithinPolygon(LocationSearchQuery query) {
+    qb.addWhereClause(
+        String.format("\"uuid\" IN (SELECT \"uuid\" FROM locations WHERE ST_Within(\"geometry\", "
+            + "ST_GeomFromText('%1$s', 3857)))", query.getWithinPolygon()));
```

Comment:
Since this code is almost the same in three places, extract it into an PostgreSQL specific) common method with two parameters.